imap/caldav_alarm.c

```diff
@@ -1271,12 +1271,79 @@ static int move_to_mailboxid(struct mailbox *srcmbox,
     return r;
 }
 
+struct find_sched_rock {
+    char *userid;
+    char *mboxname;
+    uint32_t uid;
+};
+
+static int find_sched_cb(const conv_guidrec_t *rec, void *rock)
+{
+    struct find_sched_rock *frock = (struct find_sched_rock *) rock;
+    struct buf attrib = BUF_INITIALIZER;
+    mbentry_t *mbentry = NULL;
+
```

Comment:
guid_foreach will also call the callback for deleted and expunged messages, so better the check the rec's system flags here?

Reply:
Good catch